### ISSUE TITLE
Random seed doesn't create uniform umap across machines

### ISSUE THREAD

**Issue by AceTylercholine** [EricKenjiLee/WaveMAP_Paper#4]
Following the steps from the WaveMAP paper, I can easily reproduce the same UMAP configuration because of the random seed step, however, I get a different configuration when I perform the same script in the exact same environment on a different machine. I'm not sure how this would be solved; I'm assuming there's some other background randomization process that is vaguely machine-specific.

**Comment by EricKenjiLee** [comment]
Yes there are random seeds at many different levels and they can all potentially produce a different embedding. Not sure if you've seen it but here's the code snippet that I've used that was able to preserve the same embedding across several machines I tested. Let me know if you're still seeing issues after this one. The machine specificity can be an issue for exact replication of a figure but because of the nature of the UMAP embedding process and the fact that WaveMAP clusters in high-d first, it shouldn't matter to your analysis. If you really really want exact replication, you can get a lot closer between machines by increasing the negative sampling I believe but this will really slow down the computation.

```
#Setting of random seed across Python kernel and packages to ensure reproducibility 
RAND_STATE=42
np.random.seed(RAND_STATE)
os.environ['PYTHONHASHSEED'] = str(RAND_STATE)
random.seed(RAND_STATE)
```

**Comment by AceTylercholine** [comment]
Thanks so much for your fast reply! Yes, I saw that in your helper_functions.py and used it, and I've used it like this:
`from wavemap_paper.helper_functions import RAND_STATE, set_rand_state`
`set_rand_state(RAND_STATE)`

as well as just manually inserting:
`RAND_STATE=42`
`np.random.seed(RAND_STATE)`
`os.environ['PYTHONHASHSEED'] = str(RAND_STATE)`
`random.seed(RAND_STATE)`

I've also made sure my environment/packages were the exact same. Both machines are using Windows 11. Of course, my plots are similar, I was just hoping for the results to be fully replicable. These are the plots from the same script on the same data in the same environment on my two machines:
![Megatron_UMAP](https://github.com/user-attachments/assets/e883362e-c71d-41e5-8225-b1b3056f5ee5)
![Cerebro_UMAP](https://github.com/user-attachments/assets/039d8931-5383-4f7a-a6d7-845889bb8aa8)

I think I'll try increasing the negative sampling, that's a great idea, thanks!
